services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -863,12 +925,43 @@ private Runnable getRunnableForDeferredVersionSwap() {
             break;
           }
 
+          ThreadPoolExecutor clusterExecutorService = getOrCreateExecutorForCluster(cluster);
+          ThreadPoolStats clusterThreadPoolStats = clusterToThreadPoolStatsMap.get(cluster);
+
+          // Filter out stores that are already being processed to prevent duplicate processing
+          List<Store> eligibleStoresToProcess = new ArrayList<>();
+          for (Store parentStore: parentStores) {
+            if (tryStartProcessingStore(parentStore.getName())) {
+              eligibleStoresToProcess.add(parentStore);
+            } else {
+              String message = "Skipping store " + parentStore.getName() + " as it's already being processed";
+              logMessageIfNotRedundant(message);
+            }
+          }
+
           boolean sequentialRollForward = !StringUtils.isEmpty(rolloutOrderStr);
           if (sequentialRollForward) {
-            performSequentialRollForward(cluster, parentStores, childControllerClientMap, rolloutOrder);
+            for (Store parentStore: eligibleStoresToProcess) {
+              clusterExecutorService.submit(() -> {
+                try {
```

Comment:
nit: move the `sequentialRollForward` check inside the loop to reduce duplicate codes ?